pybaseball/utils.py

```diff
@@ -48,6 +48,32 @@ def validate_datestring(date_text: Optional[str]) -> date:
     except (AssertionError, ValueError):
         raise ValueError("Incorrect data format, should be YYYY-MM-DD")
 
+def date_range(start: date, stop: date, step: int, verbose: bool = True) -> Iterator[Tuple[date, date]]:
+    '''
+    Iterate over dates. Skip the offseason dates. Returns a pair of dates for beginning and end of each segment.
+    Range is inclusive of the stop date.
+    If verbose is enabled, it will print a message if it skips offseason dates.
+    '''
+
+    low = start
+
+    while low <= stop:
+        if (low.month, low.day) <= (3, 15):
+            low = low.replace(month=3, day=15)
+            if verbose:
+                print('Skipping offseason dates')
+        elif (low.month, low.day) >= (11, 15):
+            low = low.replace(month=3, day=15, year=low.year + 1)
```

Comment:
Maybe it makes sense to add some sort of catch for 2020 here? Else it will iterate from March to July with a bunch of empty queries.

Reply:
True

Reply:
or use the actual start and end date for each season instead of always using march 15 and november 15

Reply:
Do you know where I can get that? At least for since 2008, which is when the Statcast search starts (based on PitchFX I believe?)

Reply:
If we can put this in as a hardcoded dict, that'd be ideal.

Reply:
I think we'd just have to find opening day/final day for each season manually and add a dict.

Reply:
Looks like BRef has a page we could scrape once and add to a dict (or CSV):

https://www.baseball-reference.com/leagues/MLB/2019-schedule.shtml

Reply:
Dict. I want this to be cheap enough to be used as a normal iterator.

Reply:
https://github.com/jldbc/pybaseball/pull/165